pkg/cmd/pr/create/create.go

```diff
@@ -574,6 +577,23 @@ func NewCreateContext(opts *CreateOptions) (*CreateContext, error) {
 		}
 	}
 
+	if headRepo == nil && isPushEnabled && opts.PushToBaseRepo {
+		pushableRepos, err := repoContext.HeadRepos()
+		if err != nil {
+			return nil, err
+		}
+
+		for _, r := range pushableRepos {
+			if ghrepo.IsSame(baseRepo, r) {
+				headRepo = r
+			}
+		}
+
+		if headRepo == nil {
+			return nil, fmt.Errorf("the user might lack write access to the base repository")
+		}
+	}
+
```

Comment:
Should this be refactored into the following section to avoid duplicating logic but leveraging `(r *ResolvedRemotes) RemoteForRepo`, which is doing similar logic already?

https://github.com/cli/cli/blob/3443a752a9949ee4f954a84f7ab94a4b99df1b27/context/context.go#L161-L169